includes/class-sensei-course.php

```diff
@@ -2178,9 +2178,12 @@ public function the_course_meta( $course_id ) {
 
 		if ( isset( Sensei()->settings->settings['course_author'] ) && ( Sensei()->settings->settings['course_author'] ) ) {
 			echo '<span class="course-author">' .
-				esc_html( 'by ', 'sensei-lms' ) .
-				'<a href="' . esc_attr( get_author_posts_url( $course->post_author ) ) . '" title="' . esc_attr( $author_display_name ) . '">' . esc_attr( $author_display_name ) . '</a>
-			</span>';
+				sprintf(
+					// translators: Placeholder %s is the author link.
+					esc_html__( 'by %s', 'sensei-lms' ),
+					'<a href="' . esc_attr( get_author_posts_url( $course->post_author ) ) . '" title="' . esc_attr( $author_display_name ) . '">' . esc_attr( $author_display_name ) . '</a>'
```

Comment:
Similar to https://github.com/Automattic/sensei/pull/3145/files/28d8ba21036dbf2123009f73a8df33eab7ed109a#diff-c0922c04ec6aa809fbd289ed7aabc9d7, would it make sense to flip the calls to `sprintf` and `esc_html`?

Also, it's better to embed the HTML in the translated string so that the translator has context. That would mean doing something like this:
```
'by <a href="%1$s" title="%2$s">%3$s</a>'
```

I really like [this resource](https://developer.wordpress.com/2015/04/23/wordpress-developers-test-your-i18n-internationalization-knowledge/) for figuring out how to do i18n right. Scroll down to the "You need to include a link in a sentence. How can you do that?" question.

Reply:
Flipping the call to `sprintf` and `esc_html` here is wrong because the `<a>` tag will be escaped. Do you mean using `wp_kses()` or `wp_kses_post()` instead of `esc_html()`? As I know, these functions are not good for performance.

Embedding the HTML in the translated string is good. That's a great resource. Thanks!

Reply:
OK, so in these situations we leverage `wp_kses`, but tweak the allowed HTML / attributes. An example of this can be seen [here](https://github.com/Automattic/sensei/blob/master/includes/class-sensei-settings-api.php#L519).

Reply:
Since the HTML tag is translatable, users can add more attributes than `href` and `title`. I'm thinking about allowing these ones: `href`, `title`, 'target`, `rel`, `class`. Do you think it is good enough?

Reply:
Users can also add the `<span>` tag here. Should we restrict to add other HTML tags?

Reply:
I think we should only allow the elements and attributes that we use. We could open it up later if we receive requests for it.